### PR TITLE
[STRMCMP-996] Adding translations for S3 and kinesis input sources

### PR DESCRIPTION
**Pull request by premsantosh**
Adding new translations for the S3 and kinesis source.
Also adding the composite transform that can be used to construct the S3 and Kinesis source.
------------------------
Once I build a version and tag it ill add a link here to an example running job.
**EDIT:** Example running job http://beamperfk8s5.ingress.semistateful-stg-0.us-east-1.k8s.lyft.net/#/job/55e396d43a225d0679e59c44b03cf018/overview

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

File: runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java [hunk 1]
```diff
@@ -357,6 +375,216 @@ private void translateKinesisInput(
...
+          }
+        }
+      } catch (DateTimeParseException | NumberFormatException e) {
+        // skip this event
+        LOG.warn("Skipping event: " + event.get(EventField.EventName.fieldName()));
+      }
+      return occurredAtMs;
+    }
+
+    private static long parseDateTime(String datetime) {
```
Comment (maghamravi) [inline runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java:562]: recommend: I feel we should have a utility class (EventUtils?) with such helper methods to promote code reuse . I see a similar code at  (https://github.com/lyft/beam/blob/bd762d7f35b5b4f458e129b9ba719dcd5ba00299/runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java#L652)

File: runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java [hunk 2]
```diff
@@ -357,6 +375,216 @@ private void translateKinesisInput(
...
+      }
+    }
+
+    private byte[] getBytes(Event event) {
+      ObjectMapper mapper = new ObjectMapper();
+      JsonNode eventJson = mapper.valueToTree(event);
+      ObjectWriter writer = mapper.writer();
+      byte[] bytes = new byte[0];
+      try {
+        bytes = writer.writeValueAsBytes(eventJson);
```
Comment (maghamravi) [inline runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java]: nit: can this be simplified to `mapper.writeValueAsBytes(event)`

**Comment by maghamravi** [comment]
@premsantosh Can you add few tests for the new functionality.

File: runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java (hunk 1 above)
Comment (premsantosh) [reply runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java:562]: I agree, I was looking to use the parsing logic in EventUtils initially but it looks like its not currently serializable. I believe that would require some work so I've created https://jira.lyft.net/browse/STRMCMP-1109.

**Comment by premsantosh** [comment]
> @premsantosh Can you add few tests for the new functionality.

Yep, I've added tests for the java translations. Ill look to see if I can add anything for the PTransform on the python side but I don't believe there's much to test there. I was first trying to get a working version again before adding tests. I've also added a link to a test beam job.

**Comment by maghamravi** [comment]
LGTM ! Nice work!!

File: runners/flink/flink_runner.gradle [hunk 3]
```diff
@@ -152,6 +152,7 @@ dependencies {
   compile "org.apache.flink:flink-metrics-core:$flink_version"
   compile "org.apache.flink:flink-java:$flink_version"
   compile "org.apache.flink:flink-runtime_2.11:$flink_version"
+  compile("com.lyft:streamingplatform-events-source:2.1-dryft-SNAPSHOT")
```
Comment (tweise) [inline runners/flink/flink_runner.gradle:155]: This should be further down in the lyft specific block.

Comment (tweise) [reply runners/flink/flink_runner.gradle:155]: Also, the size of the job server has grown significantly with this addition. Have you checked that we are only pulling in the minimum transitive dependencies?

**Comment by tweise** [comment]
Commits should be squashed and tagged according to the existing convention. This is to reduce friction for upgrades.

**Pull request merged**